webui_runner/benchmarks/shootout/xblabla20.c

```diff
@@ -0,0 +1,247 @@
+    }
+    st[8]  = 0x2ae36e593e46ad5f;
+    st[9]  = 0xb68f143029225fc9;
+    st[10] = 0x8da1e08468303aa6;
+    st[11] = 0xa48a209acd50a4a7;
+    st[12] = 0x7fdc12f23f90778c;
+    st[13] = 1;
+    st[14] = LOAD64_LE(&nonce[8 * 0]);
+    st[15] = LOAD64_LE(&nonce[8 * 1]);
+}
+
+static int
+blabla20_xor(uint8_t *c, const uint8_t *m, size_t len, const uint8_t nonce[blabla20_NONCEBYTES],
+             const uint8_t key[blabla20_KEYBYTES])
+{
+    uint8_t  tmp[128];
+    uint64_t ks[16];
+    uint64_t st[16];
+    uint64_t x;
+    int      i;
+
+    blabla20_init(st, nonce, key);
+    while (len >= 128) {
+        blabla20_update(ks, st);
+        for (i = 0; i < 16; i++) {
+            x = ks[i] ^ LOAD64_LE(m + 8 * i);
+            STORE64_LE(c + 8 * i, x);
+        }
+        c += 128;
+        m += 128;
+        len -= 128;
+    }
+    if (len > 0) {
+        blabla20_update(ks, st);
+        memset(tmp, 0, 128);
+        for (i = 0; i < (int) len; i++) {
+            tmp[i] = m[i];
+        }
+        for (i = 0; i < 16; i++) {
+            x = ks[i] ^ LOAD64_LE(tmp + 8 * i);
+            STORE64_LE(tmp + 8 * i, x);
+        }
+        for (i = 0; i < (int) len; i++) {
+            c[i] = tmp[i];
+        }
+    }
+    return 0;
+}
+
+static void
+hblabla20(uint8_t subkey[hblabla20_BYTES], const uint8_t nonce[hblabla20_NONCEBYTES],
+          const uint8_t key[hblabla20_KEYBYTES])
+{
+    uint64_t st[16];
+    int      i;
+
+    blabla20_init(st, &nonce[8], key);
+    st[13] = LOAD64_LE(&nonce[0]);
+    blabla20_rounds(st);
+    for (i = 0; i < 2; i++) {
+        STORE64_LE(subkey + 8 * i, st[i]);
+    }
+    for (; i < 4; i++) {
+        STORE64_LE(subkey + 8 * i, st[i + 12 - 4]);
+    }
+}
+
+static int
+xblabla20_xor(uint8_t *c, const uint8_t *m, size_t len, const uint8_t nonce[xblabla20_NONCEBYTES],
+              const uint8_t key[xblabla20_KEYBYTES])
+{
+    uint8_t subkey[blabla20_KEYBYTES];
+    uint8_t subnonce[blabla20_NONCEBYTES];
+
+    hblabla20(subkey, nonce, key);
+
+    return blabla20_xor(c, m, len, subnonce, subkey);
+}
+
+static void
+xblabla20_setup(void *global_ctx, void **ctx_p)
+{
+    (void) global_ctx;
+
+    uint8_t *buf = calloc(BUF_SIZE, (size_t) 1U);
+    assert(buf != NULL);
+
+    *ctx_p = (void *) buf;
+}
+
+static void
+xblabla20_teardown(void *ctx)
+{
+    free(ctx);
+}
+
+#ifdef WASM_ENTRY
+#ifdef ALT_ENTRY
+void
+_call(int in)
+#else
+void
+_start(void)
+#endif
+#else  // WASM_ENTRY
+#ifdef EMBED_ENTRY
+void
+xblabla20_body(void *ctx_)
+#else 
+int
+main(void)
+#endif // EMBED_ENTRY
+#endif // WASM_ENTRY
+{
+    uint8_t *buf;
+    xblabla20_setup(NULL, (void **) &buf);
+
+    BLACK_BOX(buf);
+#ifndef NO_WASI_SUPPORT
+    assert(BUF_SIZE >= xblabla20_KEYBYTES && BUF_SIZE >= xblabla20_NONCEBYTES);
+#endif
+    int i;
+    for (i = 0; i < ITERATIONS; i++) {
+        xblabla20_xor(buf, buf, BUF_SIZE, buf, buf);
+    }
+    BLACK_BOX(buf);
+    xblabla20_teardown(buf);
+}
```

Comment:
It seems like duplicating the shootout benchmarks into this lower directory could cause some confusion (e.g. "which one is the right one to use?"). Is there a way we can get to only one source file for each benchmark? If it is possible, we could open a separate issue for that.